tests/logprob/test_order.py

```diff
@@ -0,0 +1,148 @@
+#   SOFTWARE.
+
+import re
+
+import numpy as np
+import pytensor.tensor as pt
+import pytest
+
+import pymc as pm
+
+from pymc import logp
+from pymc.testing import assert_no_rvs
+
+
+def test_max():
+    """Test whether the logprob for ```pt.max``` is implemented"""
+    x = pt.random.normal(0, 1, size=(3,))
+    x.name = "x"
+    x_max = pt.max(x, axis=-1)
+    x_max_value = pt.vector("x_max_value")
+    x_max_logprob = logp(x_max, x_max_value)
+
+    assert_no_rvs(x_max_logprob)
+
+
+@pytest.mark.parametrize(
+    "x, axis",
+    [
+        (pt.random.normal(0, 1), -1),
+        (pt.random.normal(0, 1), None),
+    ],
+)
+def test_axis_max(x, axis):
+    """Test whether the rewrite takes into account ```None``` axis"""
+    x_max = pt.max(x, axis)
+    x_max_value = pt.vector("x_max_value")
+    x_max_logprob = logp(x_max, x_max_value)
+
+    assert_no_rvs(x_max_logprob)
+
+
+def test_argmax():
+    """Test whether the logprob for ```pt.argmax``` is rejected correctly"""
+    x = pt.random.normal(0, 1, size=(3,))
+    x.name = "x"
+    x_max = pt.argmax(x, axis=-1)
+    x_max_value = pt.vector("x_max_value")
+
+    with pytest.raises(RuntimeError, match=re.escape("Logprob method not implemented for Argmax")):
+        x_max_logprob = logp(x_max, x_max_value)
+
+
+def test_max_non_iid_fails():
+    """Test whether the logprob for ```pt.max``` for non i.i.d is rejected correctly"""
+    x = pm.Normal.dist([0, 1, 2, 3, 4], 1, shape=(5,))
+    x.name = "x"
+    x_max = pt.max(x, axis=-1)
+    x_max_value = pt.vector("x_max_value")
+    with pytest.raises(RuntimeError, match=re.escape("Logprob method not implemented")):
+        x_max_logprob = logp(x_max, x_max_value)
+
+
+def test_max_non_rv_fails():
+    """Test whether the logprob for ```pt.max``` for non RVs is rejected correctly"""
+    x = pt.exp(pt.random.beta(0, 1, size=(3,)))
+    x.name = "x"
+    x_max = pt.max(x, axis=-1)
+    x_max_value = pt.vector("x_max_value")
+    with pytest.raises(RuntimeError, match=re.escape("Logprob method not implemented")):
+        x_max_logprob = logp(x_max, x_max_value)
+
+
+def test_max_categorical():
+    """Test whether the logprob for ```pt.max``` for unsupported distributions is rejected correctly"""
+    x = pm.Categorical.dist([1, 1, 1, 1], shape=(5,))
+    x.name = "x"
+    x_max = pt.max(x, axis=-1)
+    x_max_value = pt.vector("x_max_value")
+    with pytest.raises(RuntimeError, match=re.escape("Logprob method not implemented")):
+        x_max_logprob = logp(x_max, x_max_value)
+
+
+def test_non_supp_axis_max():
+    """Test whether the logprob for ```pt.max``` for unsupported axis is rejected correctly"""
+    x = pt.random.normal(0, 1, size=(3, 3))
+    x.name = "x"
+    x_max = pt.max(x, axis=-1)
+    x_max_value = pt.vector("x_max_value")
+    with pytest.raises(RuntimeError, match=re.escape("Logprob method not implemented")):
+        x_max_logprob = logp(x_max, x_max_value)
+
+
+def test_max_logprob():
```

Comment:
- Perhaps parametrize this for several `test_value`s? Something like 0.01, 0.2, 0.5, 0.85, 0.9. It wouldn't hurt to check for more values.
- You can also parametrize the `n` here (which has a value of 3). This would be more important than the comment above.